lib/metric-config-parser/metric_config_parser/tests/test_experiment.py

```diff
@@ -267,6 +267,88 @@ def test_preenrollment(self, experiments, config_collection):
         assert len(days28_metrics) == 1
         assert days28_metrics[0].metric.name == "spam"
 
+    def test_segment_resolution(self, experiments, config_collection):
+        spec = AnalysisSpec()
+
+        cfg = spec.resolve(experiments[9], config_collection)
+
+        assert len(cfg.experiment.segments) == 1
+
+        segment = cfg.experiment.segments[0]
+
+        assert segment.name == "regular_users_v3"
+        assert segment.data_source.name == "my_cool_data_source"
+        expected_expression = "SELECT 1 WHERE submission_date BETWEEN"
+        assert expected_expression in segment.data_source.from_expression
```

Comment:
I think you should be able to assert the full expression here to ensure that it resolves the templated parameters properly, something like (may have gotten the dates wrong):
```suggestion
        expected_expression = "SELECT 1 WHERE submission_date BETWEEN '2019-12-01' AND '2019-12-08"
        assert expected_expression == segment.data_source.from_expression
```

Reply:
Also same comment for the other instances of this assertion.